xml_converter/src/xml_converter.cpp

```diff
@@ -62,48 +62,62 @@ vector<string> get_files_by_suffix(string directory, string suffix) {
     return files;
 }
 
-set<string> read_taco_directory(
+map<string, vector<string>> read_taco_directory(
     string input_path,
     map<string, Category>* marker_categories,
     vector<Parseable*>* parsed_pois) {
-    set<string> top_level_categories;
+    map<string, vector<string>> category_to_filepath;
     if (!filesystem::exists(input_path)) {
         cout << "Error: " << input_path << " is not an existing directory or file" << endl;
     }
     else if (filesystem::is_directory(input_path)) {
+        string directory_name = filesystem::path(input_path).filename();
         vector<string> xml_files = get_files_by_suffix(input_path, ".xml");
         for (const string& path : xml_files) {
             set<string> category_names = (parse_xml_file(path, input_path, marker_categories, parsed_pois));
-            top_level_categories.insert(category_names.begin(), category_names.end());
+            string relative_path = directory_name + path.substr(input_path.size());
```

Comment:
Can we guarantee that we are actually getting the correct relative path here and not just stripping unrelated characters if for some reason `get_files_by_suffix()` changes it's output.

Reply:
Good point. I'll add a check to ensure that the output from `get_files_by_suffix()` continues to give me the format I expect.

Reply:
Changing the function get_files_by_suffix() to return the relative paths from the directory so that we can ensure it is the correct string.